CONFIGURATION.md

```diff
@@ -0,0 +1,182 @@
+  [ http: <http_probe> ]
+
+  [ tcp: <tcp_probe> ]
+
+  [ dns: <dns_probe> ]
+
+  [ icmp: <icmp_probe> ]
+
+```
+
+### http_probe
+```yml
+
+  # Accepted status codes for this probe. Defaults to 2xx.
+  [ valid_status_codes: [ <string>, ... ] | default = "2xx" ]
+
+  # The HTTP method the probe will use.
+  [ method: [ <string> | default = "GET" ] ]
+
+  # The HTTP headers set for the probe.
+  headers:
+    [ <string>: <string>, ... ]
+
+  # Whether or not the probe will follow any redirects.
+  [ no_follow_redirects: [ <boolean> | default = false ] ]
+
+  # Probe fails if SSL is present.
+  [ fail_if_ssl: [ <boolean> | default = false ] ]
+
+  # Probe fails if SSL is not present.
+  [ fail_if_not_ssl: [ <boolean> | default = false ] ]
+
+  # Probe fails if response matches regex.
+  [ fail_if_matches_regexp: [ <regex>, ... ] | default = false ]
+
+  # Probe fails if response does not match regex
+  [ fail_if_not_matches_regexp: [ <regex>, ... ] default = false ]
+
+  # Configuration for TLS protocol of HTTP probe.
+  tls_config:
+    [ - <tls_config> ... ]
+
+  # The HTTP basic authentication credentials for the targets.
+  basic_auth:
+    [ username: [ <string> ] ]
+    [ password: [ <secret> ] ]
+
+    # The bearer token for the targets.
+    [ bearer_token: [ <secret> ] ]
+
+    # The bearer token file for the targets.
+    [ bearer_token_file: [ <filename> ] ]
+
+    # HTTP proxy server to use to connect to the targets.
+    [ proxy_url: [ <string> ] ]
+
+  # The preferred IP protocol of the HTTP probe (ip4, ip6).
+  preferred_ip_protocol: [ <string> ]
+
+  # The body of the HTTP request used in probe.
+  body: [ <string> ]
+
+
+```
+
+### tcp_probe
+
+```yml
+
+# The preferred IP protocol of the TCP probe (ip4, ip6).
+preferred_ip_protocol: [ <string> ]
+
+# The query sent in the TCP probe and the expected associated response.
+query_response:
+  [ [ [ expect: <string> ], [ send: <string> ] ], ... ]
+
+# Whether or not TLS is used.
+[ tls: [ <boolean | default = false> ] ]
+
+# Configuration for TLS protocol of TCP probe.
+tls_config:
+  [ - <tls_config> ... ]
+
+```
+
+### dns_probe
+
+```yml
+
+# The preferred IP protocol of the DNS probe (ip4, ip6).
+preferred_ip_protocol: [ <string> ]
+
+transport_protocol: [ <string> ]
+
+query_name: [ <string> ]
+
+[ query_type: [ <string> | default = "ANY" ] ]
+
+[ valid_rcodes: [ <string> | default = "NOERROR" ] ]
+
+validate_answer_rrs:
+
+  fail_if_matches_regexp:
+    [ <regex>, ... ]
+
+  fail_if_not_matches_regexp:
+    [ <regex>, ... ]
+
+validate_authority_rrs:
+
+  fail_if_matches_regexp:
+    [ <regex>, ... ]
+
+  fail_if_not_matches_regexp:
+    [ <regex>, ... ]
+
+validate_additional_rrs:
+
+  fail_if_matches_regexp:
+    [ <regex>, ... ]
+
+  fail_if_not_matches_regexp:
+    [ <regex>, ... ]
+
+```
+
+### icmp_probe
+
+```yml
+
+# The preferred IP protocol of the ICMP probe (ip4, ip6).
+preferred_ip_protocol: <string | default = "ip6" >
+
+```
+
+### tls_config
```

Comment:
This should look the same as https://prometheus.io/docs/operating/configuration/#<tls_config>